pkg/csi/service/wcp/controller.go

```diff
@@ -800,6 +800,17 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
 
 		vmuuid, err := getVMUUIDFromK8sCloudOperatorService(ctx, req.VolumeId, req.NodeId)
 		if err != nil {
+			if e, ok := status.FromError(err); ok {
+				switch e.Code() {
+				case codes.NotFound:
+					return nil, csifault.CSIVmUuidNotFoundFault, logger.LogNewErrorCodef(log, codes.Internal,
+						"failed to find the pod vmuuid annotation from the k8sCloudOperator service "+
+							"when processing attach for volumeID: %s on node: %s. Error: %+v",
+						req.VolumeId, req.NodeId, err)
+				default:
```

Comment:
nit: Do we need a default?

Reply:
Right, we don't need it when not returning error from default. Removed